server.go

```diff
@@ -429,7 +429,9 @@ type Server struct {
 	mu   sync.Mutex
 	open int32
 	stop int32
-	done chan struct{}
+
+	done      chan struct{}
+	closeDone sync.Once
```

Comment:
I  think  `atomic`  is  better than sync.Once

Reply:
How should I use atomic in this context? Like to store the state (open / closed)?